contracts/Pool.sol

```diff
@@ -418,6 +454,25 @@ contract Pool is PoolFDT {
         _emitBalanceUpdatedEvent();
     }
 
+    /**
+        @dev View function to indicate if recipient is allowed to receive a transfer.
+             This is only possible if they have zero cooldown or they are passed their withdraw window.
+    */
+    function isReceiveAllowed(address to) public view returns (bool) {
+        (uint256 lpCooldownPeriod, uint256 lpWithdrawWindow) = globals.getLpParameters();
+        return block.timestamp > (withdrawCooldown[to] + lpCooldownPeriod + lpWithdrawWindow);
+    }
+
+    /**
+        @dev Performs all necessary checks for a `transfer` call.
+    */
+    function _checkTransfer(address to, address from, uint256 wad ) internal view {
+        require(depositDate[from].add(lockupPeriod) <= block.timestamp,  "P:FUNDS_LOCKED");            // Restrict transfer during lockup period
+        require(balanceOf(from).sub(wad) >= totalCustodyAllowance[from], "P:INSUF_TRANSFERABLE_BAL");  // User can only transfer tokens that aren't custodied
+        require(isReceiveAllowed(to),                                    "P:RECIPIENT_NOT_ALLOWED");   // Recipient must not be currently withdrawing
+        require(recognizableLossesOf(from) == uint256(0),                "P:RECOG_LOSSES");            // If an LP has unrecognized losses, they must recognize losses through withdraw                                          // Update deposit date of recipient
```

Comment:
`// Update deposit date of recipient` this comment needs to be moved to be beside `_updateStakeDate`